### PR TITLE
Add logic to handle if repo's host returns 404

### PR DESCRIPTION
**Pull request by IsaacMilarky**
**Description**
- Change collection to set repo to 'Error' when repo can't be reached by reason of 404

**Signed commits**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: augur/tasks/github/detect_move/core.py [hunk 1]
```diff
@@ -44,6 +60,23 @@ def ping_github_for_repo_move(augur_db, key_auth, repo, logger,collection_hook='
         logger.warning(f"Could not check if repo moved because the api timed out 10 times. Url: {url}")
         return
 
+    
+    
+    #Mark as errored if not found
+    if response_from_gh.status_code == 404:
+        logger.error(f"Repo {repo.repo_git} responded 404 when pinged!")
+
```
Comment (sgoggins) [inline augur/tasks/github/detect_move/core.py:68]: When we get 404's we should probably do at least one retry before updating the description.

File: augur/tasks/github/detect_move/core.py [hunk 2]
```diff
@@ -44,6 +60,23 @@ def ping_github_for_repo_move(augur_db, key_auth, repo, logger,collection_hook='
...
+        repo_update_dict = {
+        'repo_git': repo.repo_git,
+        'repo_path': None,
+        'repo_name': None,
+        'description': f"During our check for this repo on {datetime.today().strftime('%Y-%m-%d')}, a 404 error was returned. The repository does not appear to have moved. Instead, it appears to be deleted"
```
Comment (sgoggins) [inline augur/tasks/github/detect_move/core.py:73]: What is the current equivalent of setting the status to 'New' for recloning? We should probably do that as well. Maybe its in here and we don't see it?

File: augur/tasks/github/detect_move/core.py [hunk 3]
```diff
@@ -69,11 +99,7 @@ def ping_github_for_repo_move(augur_db, key_auth, repo, logger,collection_hook='
         'description': f"(Originally hosted at {url}) {old_description}"
     }
 
-    current_repo_dict.update(repo_update_dict)
-
-    result = augur_db.insert_data(current_repo_dict, Repo, ['repo_id'])
-
-    logger.info(f"Updated repo for {owner}/{name}\n")
+    update_repo_with_dict(current_repo_dict, repo_update_dict, logger,augur_db)
```
Comment (sgoggins) [inline augur/tasks/github/detect_move/core.py:102]: Maybe this is where a repo gets set to be recloned the next time?

**Review by sgoggins — COMMENTED** [review]
I made some comments.

File: augur/tasks/github/detect_move/core.py (hunk 2 above)
Comment (IsaacMilarky) [reply augur/tasks/github/detect_move/core.py:73]: detect_move is only used for non-facade tasks. Facade has it's own logic to handle if repos can't be cloned. Facade set's the status to 'Failed Clone' if we can't clone it.

File: augur/tasks/github/detect_move/core.py (hunk 3 above)
Comment (IsaacMilarky) [reply augur/tasks/github/detect_move/core.py:102]: This updates the repo url with the one provided by github but it doesn't affect facade.

File: augur/tasks/github/detect_move/core.py (hunk 1 above)
Comment (IsaacMilarky) [reply augur/tasks/github/detect_move/core.py:68]: I agree I will implement that

File: augur/tasks/github/detect_move/core.py (hunk 2 above)
Comment (sgoggins) [reply augur/tasks/github/detect_move/core.py:67]: What if its deleted after we have already cloned it and populated commits? Should we have a different status for that?

**Review by sgoggins — COMMENTED** [review]
The cases we are encountering include those where the repo has previously been cloned and counted, then goes away. How are these cases handled?

**Comment by IsaacMilarky** [comment]
> The cases we are encountering include those where the repo has previously been cloned and counted, then goes away. How are these cases handled?

Facade's status is different from the other tasks so this logic won't effect that. Right now if a repo can't be cloned or updated it has a status of "Failed Clone"

File: augur/tasks/github/detect_move/core.py (hunk 2 above)
Comment (IsaacMilarky) [reply augur/tasks/github/detect_move/core.py:67]: I don't think it makes much of a difference. We could make an extra status, something like 'Stale', to be more clear about the repo's status in the database.

Comment (sgoggins) [reply augur/tasks/github/detect_move/core.py:67]: I think the thing that's important is making sure we don't get a singular network issue that marks all the repositories some kind of "stale'.

**Review by sgoggins — APPROVED** [review]
I approve this, AND I strongly prefer adding some logic to retry so a single 1 hour network failure does not mark all the repos incorrectly stale.

**Pull request merged**